javascript-loops-practice/add-suffix-to-all.js

```diff
@@ -0,0 +1,7 @@
+/* exported addSuffixToAll */
+function addSuffixToAll(words, suffix) {
+  var newArray = [];
+  for (var i = 0; i < words.length; i++) {
+    newArray.push(words[i] += suffix);
```

Comment:
_**Uh-oh**_, this is bad news. By using the `+=` operator, you are changing the original array permanently, which is an example of something called a "side effect". Unless it is the intended purpose of the function, a function should **_never_** make changes to the content of any arrays or objects given as arguments to the function. Arrays and objects are given as arguments to share the data inside of them, not make changes to it.